src/main/java/com/pingcap/tikv/Main.java

```diff
@@ -1,10 +1,79 @@
 package com.pingcap.tikv;
 
 
+import com.google.common.collect.ImmutableList;
+import com.pingcap.tikv.catalog.Catalog;
+import com.pingcap.tikv.expression.TiColumnRef;
+import com.pingcap.tikv.expression.aggregate.Count;
+import com.pingcap.tikv.meta.TiDBInfo;
+import com.pingcap.tikv.meta.TiSelectRequest;
+import com.pingcap.tikv.meta.TiTableInfo;
+import com.pingcap.tikv.operation.SchemaInfer;
+import com.pingcap.tikv.predicates.ScanBuilder;
+import com.pingcap.tikv.row.Row;
+import com.pingcap.tikv.util.RangeSplitter;
+import com.pingcap.tikv.util.RangeSplitter.RegionTask;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Scanner;
 import org.apache.log4j.Logger;
 
 public class Main {
   private static final Logger logger = Logger.getLogger(Main.class);
   public static void main(String[] args) throws Exception {
+    TiConfiguration conf = TiConfiguration.createDefault("127.0.0.1:2379");
+    TiSession session = TiSession.create(conf);
+    Catalog cat = session.getCatalog();
+    TiDBInfo db = cat.getDatabase("test");
+    TiTableInfo table = cat.getTable(db, "t1");
+    Snapshot snapshot = session.createSnapshot();
+    TiSelectRequest selectRequest = new TiSelectRequest();
+    ScanBuilder scanBuilder = new ScanBuilder();
+    ScanBuilder.ScanPlan scanPlan = scanBuilder.buildScan(new ArrayList<>(), table);
+    selectRequest.addRequiredColumn(TiColumnRef.create("c1", table))
+        .addRequiredColumn(TiColumnRef.create("c2", table))
+        .addRequiredColumn(TiColumnRef.create("s1", table))
+        .addRequiredColumn(TiColumnRef.create("f1", table))
+        .setStartTs(snapshot.getVersion())
+        .addRanges(scanPlan.getKeyRanges())
+        .setTableInfo(table);
+    Iterator<Row> it = snapshot.tableRead(selectRequest);
```

Comment:
`snapshot.tableRead` now only accepts a `TiDAGRequest` rather than `TiSelectRequest`, which will result in complication error when `mvn package` this PR.

Reply:
DONE.